packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -519,9 +519,14 @@ class HotRunner extends ResidentRunner {
     await _evictDirtyAssets();
     printTrace('Reassembling application');
     bool reassembleAndScheduleErrors = false;
+    bool reassembleTimeout = false;
     for (FlutterView view in reassembleViews) {
       try {
         await view.uiIsolate.flutterReassemble();
+      } on TimeoutException {
+        reassembleTimeout = true;
+        printStatus("Reassembling ${view.uiIsolate.name} took too long. Hot reload may have failed.");
```

Comment:
I don't think a Flutter user would know what "reassemble" means. I suggest `printStatus` something less precise but more comprehensible for users unfamiliar with the inner workings of the hot-reload pipeline, but move the current message to `printTrace` (we could then also add more useful debugging information there).